### PR TITLE
various operator fixes

### PR DESCRIPTION
**Pull request by pohly**
This covers validation, defaulting and conversion.

### CODE REVIEW

**Comment by pohly** [comment]
I removed `validateDriver` calls in some tests because that slowed down testing and didn't fit the purpose of the test.

Now it seems that sometimes something gets stuck: https://cloudnative-k8sci.southcentralus.cloudapp.azure.com/blue/organizations/jenkins/pmem-csi/detail/PR-830/1/tests

Deleting the CR hangs and eventually times out. I saw that once also locally, but couldn't reproduce it.

Any idea why it might get stuck?

**Comment by pohly** [comment]
Nope, not related to changes in this PR. It also fails the same way in https://cloudnative-k8sci.southcentralus.cloudapp.azure.com/blue/organizations/jenkins/pmem-csi/detail/PR-825/6/tests

**Comment by avalluri** [comment]
> Nope, not related to changes in this PR. It also fails the same way in https://cloudnative-k8sci.southcentralus.cloudapp.azure.com/blue/organizations/jenkins/pmem-csi/detail/PR-825/6/tests

The reals issue is with running the earlier test:
```
 Dec 16 18:06:18.683: INFO: create deployment error: object is being deleted: deployments.pmem-csi.intel.com "alpha-default-values" already exists, will retry...
[2020-12-16T18:06:20.086Z] Dec 16 18:06:19.639: FAIL: Timed out after 180.000s.
[2020-12-16T18:06:20.087Z] create deployment "alpha-default-values"
[2020-12-16T18:06:20.087Z] Expected
[2020-12-16T18:06:20.087Z]     <*errors.StatusError | 0xc0019faa00>: {
[2020-12-16T18:06:20.087Z]         ErrStatus: {
[2020-12-16T18:06:20.087Z]             TypeMeta: {Kind: "Status", APIVersion: "v1"},
[2020-12-16T18:06:20.087Z]             ListMeta: {
[2020-12-16T18:06:20.087Z]                 SelfLink: "",
[2020-12-16T18:06:20.087Z]                 ResourceVersion: "",
[2020-12-16T18:06:20.087Z]                 Continue: "",
[2020-12-16T18:06:20.087Z]                 RemainingItemCount: nil,
[2020-12-16T18:06:20.087Z]             },
[2020-12-16T18:06:20.087Z]             Status: "Failure",
[2020-12-16T18:06:20.087Z]             Message: "object is being deleted: deployments.pmem-csi.intel.com \"alpha-default-values\" already exists",
[2020-12-16T18:06:20.087Z]             Reason: "AlreadyExists",
[2020-12-16T18:06:20.087Z]             Details: {
[2020-12-16T18:06:20.087Z]                 Name: "alpha-default-values",
[2020-12-16T18:06:20.087Z]                 Group: "pmem-csi.intel.com",
[2020-12-16T18:06:20.087Z]                 Kind: "deployments",
[2020-12-16T18:06:20.087Z]                 UID: "",
[2020-12-16T18:06:20.087Z]                 Causes: nil,
[2020-12-16T18:06:20.087Z]                 RetryAfterSeconds: 0,
[2020-12-16T18:06:20.087Z]             },
[2020-12-16T18:06:20.087Z]             Code: 409,
[2020-12-16T18:06:20.087Z]         },
[2020-12-16T18:06:20.087Z]     }
[2020-12-16T18:06:20.087Z] to be nil
```

**Comment by pohly** [comment]
> The reals issue is with running the earlier test:
[...]
Same problem, somehow CR deletion got stuck.

File: pkg/apis/pmemcsi/v1alpha1/deployment_conversion.go [hunk 1]
```diff
@@ -37,15 +37,21 @@ func (d *Deployment) ConvertTo(dst conversion.Hub) error {
 
 	// no change in other fields
 	out.ObjectMeta = in.ObjectMeta
-	out.Spec.LogLevel = in.Spec.LogLevel
 	out.Spec.Image = in.Spec.Image
-	out.Spec.CACert = in.Spec.CACert
+	out.Spec.PullPolicy = in.Spec.PullPolicy
```
Comment (avalluri) [inline pkg/apis/pmemcsi/v1alpha1/deployment_conversion.go:41]: @pohly I googled about this missing conversions and came across about [conversion-gen](https://godoc.org/k8s.io/code-generator/cmd/conversion-gen#:~:text=conversion%2Dgen%20is%20a%20tool,between%20internal%20and%20external%20types.&text=The%20generated%20conversion%20functions%20use%20standard%20value%20assignment%20wherever%20possible.) code generation tool, that generates the type conversion code as mentioned [here](https://github.com/kubernetes-sigs/kubebuilder/issues/1529#issuecomment-656359330). What do you think, shall we use that as a separate change or part of this chagne?

Comment (pohly) [reply pkg/apis/pmemcsi/v1alpha1/deployment_conversion.go:41]: If we use it, then in a separate PR.

I want to get into a stable state as quickly as possible again and using that tool may need further investigations and discussions.

**Review by avalluri — APPROVED** [review]

**Pull request merged**